### DISCUSSION TITLE
How to use k-crop and multi-scale strategies during the test phase?

### DISCUSSION THREAD

**Discussion by IncludeMathH** [open-mmlab/mmclassification#1227] [Q&A]
In [ResNet's](https://arxiv.org/abs/1512.03385) paper, the author uses 10-crop and multi-scale testing strategies. Multi-scale testing is also supported in `MMDetection` projects, but is there any in `MMClassification` ? If it is not available at present and I need to write my own program, is there a roadmap to provide it?

**Comment by IncludeMathH** [comment]
I think I found some clues in `base.py`, which I will try to achieve first.
```python
    def forward_test(self, imgs, **kwargs):
        """
        Args:
            imgs (List[Tensor]): the outer list indicates test-time
                augmentations and inner Tensor should have a shape NxCxHxW,
                which contains all images in the batch.
        """
        if isinstance(imgs, torch.Tensor):
            imgs = [imgs]
        for var, name in [(imgs, 'imgs')]:
            if not isinstance(var, list):
                raise TypeError(f'{name} must be a list, but got {type(var)}')

        if len(imgs) == 1:
            return self.simple_test(imgs[0], **kwargs)
        else:
            raise NotImplementedError('aug_test has not been implemented')
```

**Reply by yingfhu** [reply]
1. TTA is a incoming new feature, refers to https://github.com/open-mmlab/mmclassification/pull/1161 if you want it in advanced, and this will be merge into 1.x branch finally.
2. K-crop is not implemented, please customize your transforms refers to [docs in 1.x branch](https://mmclassification.readthedocs.io/en/1.x/advanced_guides/pipeline.html)

**Reply by IncludeMathH** [reply]
OK, thank you for your reply. I'll try the TTA strategy you mentioned first.
